main_data_classes.rb

```diff
@@ -0,0 +1,211 @@
+    #return sum_digits
+    puts "Your number's digits sum is #{sum_digits}"
+  else
+    #return 0
+    puts "Sorry, but #{nat_num} is not a number or not a natural number"
+  end
+end
+
+
+#2. There is a string. Find a maximum number of digits placed in a row in it
+
+#str = 'qw12e63406q4qeg156g16w1'
+def digits_maxnum_in_string(str)
+  arr = Array.new
+  max_length = 0
+  str.chars do |char|
+    if char == '0'
+      i_char = char.to_i
+      arr = arr << i_char
+      if max_length < arr.length
+        max_length = arr.length
+      end
+    elsif (i_char = char.to_i) > 0
+      arr = arr << i_char
+      if max_length < arr.length
+        max_length = arr.length
+      end
+    else
+      arr = Array.new
+    end
+  end
+  puts "Maximum number of digits in a row is #{max_length}"
+  #return max_length
+end
+
+
+#3. There is a string. Find a maximal number (not a digit) in it
+
+#str = 'qw10e614q4qeg156g16w1'
+def max_number_in_string(str)
+  arr = Array.new
+  number = 0
+  temp_num = 0
+  temp_str = ''
+  str.chars do |char|
+    if char == '0'
+      i_char = char.to_i
+      arr = arr << i_char
+      temp_str = temp_str + arr[-1].to_s
+      temp_num = temp_str.to_i
+      if number < temp_num
+        number = temp_num
+      else
+        arr = Array.new
+      end
+    elsif (i_char = char.to_i) > 0
+      arr = arr << i_char
+      temp_str = temp_str + arr[-1].to_s
+      temp_num = temp_str.to_i
+      if number < temp_num
+        number = temp_num
+      else
+        arr = Array.new
+      end
+    else
+      arr = Array.new
+      temp_num = 0
+      temp_str = ''
+    end
+  end
+  puts "Maximum number in a string is #{number}"
+  #return number
+end
+
+
+#4. There are two strings. Find number of first chars from first string matching first chars of the second string. Consider two cases:
+#- strings are definitely different
+#- strings can completely match
+#str1 = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit'
+#str2 = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit'
+#str2 = 'Lorem ipsum sit amet, consectetur adipiscing elit'
+#str2 = 'Lorem456'
+#str2 = 'orem ipsum'
+#str2 = ''
+
+def strings_matching(str1, str2)
+  arr1 = str1.split('')
+  arr2 = str2.split('')
+
+  if arr1.length <= arr2.length
+    length = arr1.length
+  else
+    length = arr2.length
+  end
+
+  number = 0
+  while number<length && arr1[number] == arr2[number]
+    number+=1
+  end
+  puts "Number is #{number}"
+  #return number
+end
+
+
+#5. There is an array of integers. First puts elements with even indexes and then with odd indexes
+
+#ary = [1,2,3,4,5,6,7,8,9]
+def even_and_odd_indexes(ary)
+  puts 'Elements with even indexes are:'
+  i = 0
+  while i < ary.length
+    puts ary[i]
+    i = i+2
+  end
+  puts 'Elements with odd indexes are:'
+  i = 1
+  while i < ary.length
+    puts ary[i]
+    i = i+2
+  end
+end
+
+
+#6. There is an array of integers (ary). Puts the index of the last element where ary[0]<ary[i]<ary[-1]
```

Comment:
what if the array would look like this:
```[3,1,5,7,8,0,2,9]```
I would expect index of element 8 as an answer